tracetools/propagate.go

```diff
@@ -0,0 +1,96 @@
+package tracetools
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/gob"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// EnvVarPropagator is a custom distributed tracing propagator that uses environment
+// variables as the transport. This can be used to implement distributed tracing
+// across shell processes where turn-key tracing solutions may not exist.
+//
+// See usage in the example test.
+//
+// Internally, this uses a text map carrier and then serializes and encodes the data
+// so that it can used as an environment variable.
+//
+// Why the carrier data into a single string? Since carrier data is just key-value
+// pairs, they could be directly added as env vars. However, when propagating the
+// env vars to new processes, users will have to 1) know the list of carrier keys
+// and 2) iterate and push them through. Since carrier keys are a bit of an
+// implementation detail, they can change easily. So containing it all in one well
+// known env var makes things simpler.
+//
+// TODO: fallback to BUILD_ID/JOB_ID?
```

Comment:
This TODO made sense when we were using ddtrace in python, but in Go the trace_id and span_id have to be int64s, which isn't compatible with `BUILD_ID/JOB_ID` since they're 128-bit UUIDs. Should be fine either way as long as people have searching by tags available to them in their tracing solution.